xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/displayer_page.vm

```diff
@@ -1 +1,40 @@
-#template('displayer_dblist.vm')
+#if ($type == 'edit' || $type == 'search')
+  #if ($field.getProperty('displayType').value == 'input' && $field.getProperty('picker').value == 1)
+    #xpropertySuggestInputDisplayer($field $prefix $name $value)
+  #else
+    $doc.displayEdit($field, $prefix, $object)
+  #end
+#elseif ($type == 'view')
+  #set ($pages = [])
+  #foreach($pageReference in $value)
+    #set ($discard = $pages.add("#displayPageReference($pageReference)"))
+  #end
+  #if ($pages.size() > 0)
```

Comment:
I would not mix the code that determines the values with the code that displays the values (the call to ``displayPageReference``). When determining the values there are 3 cases, see the ``suggestInput`` macro in [macros.vm](https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/macros.vm#L2544):

1. ``$value`` is an array (because relational storage is enabled)
1. ``$value`` is not an array and multiple selection is enabled (because relational storage is disabled)
1. ``$value`` is not an array and single selection is enabled